core/src/main/java/com/linecorp/armeria/server/Routers.java

```diff
@@ -227,7 +230,34 @@
             // Set a comparator to sort services by the number of conditions to be checked in a descending
             // order.
             builder.comparator(valueComparator);
-            values.forEach(v -> builder.add(routeResolver.apply(v).paths().get(1), v));
+            values.forEach(v -> {
+                final Route route = routeResolver.apply(v);
+                final String triePath = route.paths().get(1);
```

Comment:
How about using the first path so we can simplify the switch logic?`
```java
route.paths().get(0); // This will be "/foo/" when the type is PREFIX.
```

Reply:
Simplified